src/main/java/com/autosale/shop/service/impl/AuthenticationServiceImpl.java

```diff
@@ -22,37 +22,39 @@ public class AuthenticationServiceImpl implements AuthenticationService {
 
     private final JwtTokenService jwtTokenService;
     private final UserService userService;
+    private final UserSessionService userSessionService;
 
     @Override
     public JwtTokensDTO generateTokensByUserCredentials(User user) {
-
         user = userService.getVerifiedUser(user.getUserName(), user.getPassword());
+        userSessionService.createSession(user);
```

Comment:
session should be created in AuthenticationService. Session is not related to token generating

Reply:
actually, methods that create tokens are the only way to log in for users, this code is in AuthenticationService implementation, what should I do then?

Reply:
ok, leave it here